# Chapter 1: Python Basics
## 1.1 Using Python

In [1]:
# check if you have python installed
print("Hello, Universe!")

Hello, Universe!


## 1.2 Understanding Expression and Assignments

In [6]:
# calculate Earth velocity around the Sun
# 2*Pi*radius in km/period which is a year in seconds
print(2 * 3.14159 * (1.496* 10**8) / (3.156*10**7))
# or 
print(2 * 3.14159 * 1.496e8/ 3.156e7)

29.783388086185045
29.783388086185045


In [11]:
# make a simple calculator
radius = 1.496e8 # orbital radius in km
period = 3.156e7 # orbital period in s
pi = 3.14159     # value of pi

# calculate the orbital velocity
velocity = 2 * pi * radius / period

# display output
print(velocity)
# or
print("orbital velocity = ", velocity, " km/s")

29.783388086185045
orbital velocity =  29.783388086185045  km/s


In [12]:
# discard unimportant digits that might cause errors in computations
print("orbital velocity = {:5.2f} km/s".format(velocity))

orbital velocity = 29.78 km/s


In [13]:
# calculate the radius of an imaginary planet that is 10 times bigger than Earth
radius = 10*radius
print("the radius of the imaginary planet = {:.3e} km".format(radius))


the radius of the imaginary planet = 1.496e+09 km


### Kepler's third law of planetary motion, for a planet that orbits in a circular motion around our sun
$${P^2} ={\frac{4\pi^2}{GM}}\times{r^3}$$
where $$M = 1.989\times{10^30}kg$$ is the mass of the ☀
and $$G = 6.674\times{10^-10} N {kg^-2}{m^2}$$ is the gravitational constant
we can rewrite it as: $$P = 2\pi \times{GM^{\frac{-1}{2}}}{r^{\frac{3}{2}}}$$



In [14]:
# calculator for period in seconds given the radius in km based on Kepler's third law 
g = 6.674e-11
m = 1.989e30
period = 2 * pi * (g * m ) ** (-1/2) * (1e3 * radius) ** (3/2) # we converted the radius from km to m

# display period on years
print("new orbital period = {:.1f} years".format(period/3.156e7)) # we converted seconds into years

velocity = 2 * pi * radius / period
# dispaly velocity
print("new orbital velocity = {:.2f} km/s".format(velocity))
# remember that this velocity belongs to that imaginary planet that is 10 times bigger than Earth!


new orbital period = 31.6 years
new orbital velocity = 9.42 km/s


### so 1 year in this planet is 31.6 of Earth years! as its slower than earth when it orbits around the ☀

## 1.3 Control Structures